narwhals/_pandas_like/dataframe.py

```diff
@@ -122,11 +122,30 @@ def __getitem__(
                 backend_version=self._backend_version,
             )
 
+        elif (
+            isinstance(item, tuple)
+            and len(item) == 2
+            and isinstance(item[1], (tuple, list))
+        ):
+            if all(isinstance(x, int) for x in item[1]):
+                return self._from_native_dataframe(self._native_dataframe.iloc[item])
+            if all(isinstance(x, str) for x in item[1]):
+                item = (
+                    item[0],
+                    self._native_dataframe.columns.get_indexer(item[1]),
+                )
+                return self._from_native_dataframe(self._native_dataframe.iloc[item])
+            msg = (
+                f"Expected sequence str or int, got: {type(item[1])}"  # pragma: no cover
+            )
+            raise TypeError(msg)  # pragma: no cover
+
         elif isinstance(item, tuple) and len(item) == 2:
             from narwhals._pandas_like.series import PandasLikeSeries
 
             if isinstance(item[1], str):
-                native_series = self._native_dataframe.loc[item]
+                item = (item[0], self._native_dataframe.columns.get_loc(item[1]))
+                native_series = self._native_dataframe.iloc[item]
```

Comment:
drive-by: this wasn't quite right to begin with. I've added an extra test case to `test_gather_pandas_index` which demonstrates the issue (which this pr fixes)

no blame of course 🤗 this is just a fyi